rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml

```diff
@@ -79,5 +79,185 @@
                 </hint>
             </perform>
         </rule>
+        <rule id="xml-renamed-components-00004">
+            <when>
+                <project>
+                    <artifact groupId="org.apache.camel" artifactId="camel-hdfs2" />
+                </project>
+            </when>
+            <perform>
+                <hint title="`org.apache.camel:camel-hdfs2` artifact has been renamed to `org.apache.camel:camel-hdfs`" effort="1" category-id="mandatory" >
+                    <message>`org.apache.camel:camel-hdfs2` artifact has been renamed to `org.apache.camel:camel-hdfs`. It’s corresponding component package was renamed from `org.apache.camel.component.hdfs2` to `org.apache.camel.component.hdfs`. The supported scheme is now `hdfs`.</message>
+                    <link title="Camel 3 - Renamed Components" href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_renamed_components" />
+                    <quickfix type="REPLACE" name="camel-hdfs2-replace">
+                        <replacement>camel-hdfs2</replacement>
+                        <search>camel-hdfs</search>
+                    </quickfix>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="xml-renamed-components-00005">
+            <when>
+                <project>
+                    <artifact groupId="org.apache.camel" artifactId="camel-hdfs" />
```

Comment:
Add the version to avoid a kind of changes chain: the previous rule suggests to change from `camel-hdfs2` to `camel-hdfs` and once the user do it and run again RHAMT, it will have this rule to fire that suggests to change the code but the suer was already using `camel-hdfs2`so no code changes are required mandatory.
Adding the version will help the user figure out that upgrading from version 2.x to 3.y the code might change.
```suggestion
                    <artifact groupId="org.apache.camel" artifactId="camel-hdfs" toVersion="2"/>
```

Reply:
Hi @mrizzi, thanks for reviewing ! When I add `toVersion="2"` to the rule as you suggested I get the following error in the tests:
`Failed tests: 
  WindupRulesMultipleTests.executeRule:170->visit:318 The rule with id="xml-renamed-components-00009" has failed with message:String index out of range: 0`

Does this sound familiar to you ? Thanks !

Reply:
Hi @johnpoth, i can see the error is coming from rule `xml-renamed-components-00009` so maybe it could something related to that rule?
If you want feel free to push the changes even if broken so that i can run it locally and reproduce the issue

Reply:
Hi @mrizzi, the error comes from the version not being set in the test pom file. Once added the error goes away and the test passes. Thanks !